src/dinitctl.cc

```diff
@@ -324,8 +345,41 @@ int dinitctl_main(int argc, char **argv)
             cmdline_error = true;
         }
     }
+    else if (command == command_t::SIG_SEND && sig_num == -1) {
+        bool converted = false;
+        try
+        {
+            if (!std::all_of(sigstr.begin(), sigstr.end(), ::isdigit) &&
+                std::any_of(sigstr.begin(), sigstr.end(), ::isdigit)) {
```

Comment:
I'm not sure about this check, Are we have a signal which have a combine of numbers and characters?
and catching `std::bad_alloc` is enough?

Reply:
Don't check it like this. What about USR1 and USR2?

`stoi` can tell you whether the input was completely converted. You'll need to pass the additional arguments to it.

I don't understand what would throw `bad_alloc` nor why you would catch it here.